### PR TITLE
Refactor: clean up obsolete backend packages and workflow runtime

### LINKED ISSUE #579: Clean up obsolete packages and deprecated code paths in backend
### Current Limitation
The `backend` directory contains several obsolete packages, files, and configurations that are either unused, half-implemented, or represent deprecated code paths after the migration to `taskv2` and newer systems. Keeping these obsolete items clutters the codebase, increases cognitive overhead, and makes maintenance harder.

Specifically, the following obsolete/deprecated items were identified:
1. **Unused Task Manager v1 (`backend/internal/task/`)**: The application bootstrap (`backend/internal/app/bootstrap/app.go`) now fully uses `taskv2` and `taskv2/plugins`. The older `backend/internal/task/` package (except for references to `task/plugin` types like `Type` and `State`) is no longer wired or utilized.
2. **Dangling package `backend/internal/taskworkflow/`**: Contains persistence code for task workflows, but is completely unused and unimported across the entire repository.
3. **Deprecated Fields/Comments**:
   - `outcome` in `backend/internal/task/manager/manager.go` (marked to be removed after workflow manager v1 is deprecated).
   - `EmittedOutcome` in `backend/internal/task/plugin/plugin.go` and `simple_form.go`.
   - Deprecated `SubmissionURL` in `simple_form.go` (prefers `Submission.Url`).

### Suggested Improvement
We should systematically remove these obsolete components one by one:
1. **Remove Unused v1 Task Manager**:
   - Verify that no external or background systems depend on the v1 task manager runtime.
   - Refactor references to `github.com/OpenNSW/nsw/backend/internal/task/plugin` (e.g. `WorkflowNodeTemplateType` in `backend/internal/workflow/model/workflow_node.go`) to use types defined elsewhere or move them to a shared package.
   - Delete `backend/internal/task/`.
2. **Remove Unused Packages**:
   - Safely delete the dangling `backend/internal/taskworkflow/` directory.
   - Clarify the status of `backend/internal/paymentsv2/` (either integrate it to replace `backend/internal/payments` or remove it if obsolete/draft).
3. **Clean Up Deprecated Code/Comments**:
   - Remove comments and deprecated fields/struct members (`outcome`, `EmittedOutcome`, `SubmissionURL`) once the transition is finalized.

### Version
Commit: 4f3c641527acbba152dbae31000592c2e3c21467

### Additional Context
A cleanup of `backend/internal/workflow/runtime/` is already underway (files like `runtime.go`, `runtime_test.go`, `upstream.go` have been staged for deletion). This request focuses on cleaning up the remaining obsolete task/workflow/payments directories and deprecated code paths to ensure a clean codebase.

### PR DESCRIPTION
**Pull request by mushrafmim**
## Summary

This PR cleans up the obsolete task manager v1, taskworkflow, form, and workflow runtime packages from the backend repository. It refactors workflow_node.go to use a simple string type for `WorkflowNodeTemplateType` instead of the obsolete `taskPlugin.Type`. It also adds a unit test for the `taskv2` registry loader and updates the config loader to correctly register top-level workflow configurations. Additionally, it removes obsolete database migrations and baseline tables associated with the deleted packages (`forms`, `task_infos`, `task_workflow_tasks`).

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

## Changes Made

- Deleted the unused v1 task manager under `backend/internal/task/` (including all container, manager, persistence, and plugin code).
- Deleted the dangling persistence code in `backend/internal/taskworkflow/`.
- Deleted the old workflow runtime files under `backend/internal/workflow/runtime/`.
- Deleted the obsolete `backend/internal/form/` package.
- Refactored `backend/internal/workflow/model/workflow_node.go` to use string representation for `WorkflowNodeTemplateType` instead of the deprecated `taskPlugin.Type`.
- Fixed `backend/internal/taskv2/registry/config_loader.go` to parse and register top-level JSON workflow files under the configuration directory.
- Added a unit test suite in `backend/internal/taskv2/registry/config_loader_test.go` to verify task folder parsing and registration.
- Removed SQL definitions and indices for obsolete tables `task_infos` and `forms` in `001_initial_schema.up.sql` / `001_initial_schema.down.sql`.
- Deleted seed data migrations `003_insert_seed_form_templates.up.sql` / `003_insert_seed_form_templates.down.sql` and `013_fcau_forms_seed.up.sql` / `013_fcau_forms_seed.down.sql` since they only target the deleted `forms` table.
- Deleted table creation migration `012_create_task_workflow_tasks.up.sql` / `012_create_task_workflow_tasks.down.sql` for the obsolete `task_workflow_tasks` table.
- Updated `run.sh` and `down.sh` to remove execution references to the deleted SQL migrations.

## Testing

- [x] I have tested this change locally
- [x] I have added unit tests for new functionality
- [x] I have tested edge cases
- [x] All existing tests pass

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Related Issues

Closes #579

## Screenshots/Demo

N/A

## Additional Notes

All backend tests pass successfully after these deletions and refactoring.

## Deployment Notes

N/A

### CODE REVIEW

File: backend/internal/taskv2/registry/config_loader.go [hunk 1]
```diff
@@ -48,13 +48,29 @@ func LoadConfigsInto(reg *InMemRegistry, rootDir string) error {
...
-		if !e.IsDir() {
-			continue
-		}
-		if err := loadTaskFolder(reg, filepath.Join(rootDir, e.Name())); err != nil {
-			return err
+		if e.IsDir() {
+			if err := loadTaskFolder(reg, filepath.Join(rootDir, e.Name())); err != nil {
+				return err
+			}
+			taskFolders++
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply backend/internal/taskv2/registry/config_loader.go]: Accidentally pushed, removed now.

File: backend/internal/taskv2/registry/config_loader.go [hunk 2]
```diff
@@ -48,13 +48,29 @@ func LoadConfigsInto(reg *InMemRegistry, rootDir string) error {
...
+				}
+				if !json.Valid(data) {
+					continue
+				}
+				var w engine.WorkflowDefinition
+				if err := json.Unmarshal(data, &w); err == nil && w.ID != "" {
+					reg.RegisterWorkflow(w)
+					slog.Info("registered top-level workflow", "id", w.ID, "path", path)
+				}
+			}
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply backend/internal/taskv2/registry/config_loader.go]: Accidentally pushed, removed now.

**Review by ginaxu1 — APPROVED** [review]

**Review by sthanikan2000 — APPROVED** [review]
